### DISCUSSION TITLE
Quarkus kubernetes/openshift/minikube extension existing resources?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#22258] [Q&A] [answered]
Following the [use existing resources section of the Quarkus Kubernetes Guide](https://quarkus.io/guides/deploying-to-kubernetes#using-existing-resources) it mentions I can place files in `src/main/kubernetes` that will be picked up by the extensions.

It says

   > Those resources can be added under src/main/kubernetes directory and can be named after the target environment (e.g. kubernetes.json, openshift.json, knative.json, or the yml equivalents). Each of these files may contain one or more Kubernetes resources.

What if I have some resources that I want added to the generated `kubernetes.yml` and `openshift.yml`? It seems that only specific filenames of `kubernetes.yml` or `openshift.yml` are picked up in `src/main/kubernetes`. What if I have a resource, like a `ConfigMap`, that I want to appear in all the generated descriptors? Do I need to create both a `src/main/kubernetes/kubernetes.yml` and a `src/main/kubernetes/openshift.yml` with the same content?

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
@edeandrea indeed you do. @iocanel maybe it makes sense to improve on that somehow

**Reply by iocanel** [reply]
It does! Maybe each file that does not correspond to a known target spells be tested as common. Or maybe we should use an specific mama like 'all' or 'common'.

**Reply by geoand** [reply]
I like the common idea

**Reply by edeandrea** [reply]
You use the word "base" in the guide. Maybe use that?

Common works too.

The jkube/fabric8 maven plugins allow you to use things like `deployment`/`configmap`/`deploymentconfig`/etc.

**Reply by edeandrea** [reply]
Or even any name other than one of the ones it's looking for today

**Reply by edeandrea** [reply]
I created #22290 from this
